src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/CommentsIndentationCheck.java

```diff
@@ -0,0 +1,496 @@
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+package com.puppycrawl.tools.checkstyle.checks.indentation;
+
+import com.puppycrawl.tools.checkstyle.api.Check;
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+/**
+ * This Check controls the indentation inside comments and between comments and surrounding code.
+ * There're should be a whitespace between comment declaration (//, /*)
+ * and it's content. Single-line block comments (/* ... *&#47;)
+ * have to have a whitespace before the {@link TokenTypes#BLOCK_COMMENT_END block comment end}
+ * (*&#47;)<br>
+ * Comments are indented at the same level as the surrounding code.
+ * Detailed info about such convention can be found
+ * <a href=
+ * "http://google-styleguide.googlecode.com/svn/trunk/javaguide.html#s4.8.6.1-block-comment-style">
+ * here</a>
+ * <p>
+ * Examples:
+ * </p>
+ * <p>
+ * Check's config:
+ * </p>
+ * <p>
+ * <code>
+ * &lt;module name=&quot;CommentsIndentation&quot;/&gt;
+ * </code>
+ * </p>
+ * <p>
+ * <pre>
+ * <code>
+ * //  comment - this comment has incorrect indentation level, expected 1, but was 2.
+ * //comment - expected 1, but was 0.
+ * /*comment *&#47; - expected 1, but was 0.
+ * /* comment*&#47; - Block comment end should be preceded with a whitespace.
+ * /*
+ * comment; - has no asterisk at the beginning
+ *  *comment; - is not indented from the beginning
+ *  * that comment is ok. - OK
+ *  *&#47;<br>
+ * /*
+ *  *
+ *  * comment*&#47; - Block comment end should be preceded with a whitespace.
+ * </code>
+ * </pre>
+ * <pre>
+ * <code>
+ * /*
+ *  * comment
+ *  * some comment
+ *  *&#47;
+ * boolean bool = true; - such comment indentation is ok
+ *    /*
+ *    * comment
+ *    * some comment
+ *     *&#47;
+ * double d = 3.14; - Block Comment has incorrect indentation level 7, expected 4.
+ * // some comment - comment is ok
+ * String str = "";
+ *     // some comment Comment has incorrect indentation level 8, expected 4.
+ * String str1 = "";
+ * </code>
+ * </pre>
+ * </p>
+ *
+ * @author <a href="mailto:nesterenko-aleksey@list.ru">Aleksey Nesterenko</a>
+ *
+ */
+public class CommentsIndentationCheck extends Check
+{
+
+    /**
+     * A key is pointing to the warning message text in "messages.properties"
+     * file.
+     */
+    public static final String MSG_KEY_COMMENTS_INDENTATION = "comments.indentation";
```

Comment:
Just one-line `Key pointing to the warning message text in "messages.properties" file.` would be enough